src/ui/hooks/useAccountAvailable.ts

```diff
@@ -0,0 +1,16 @@
+// Copyright 2022-2024 @paritytech/contracts-ui authors & contributors
+// SPDX-License-Identifier: GPL-3.0-only
+
+import { keyring } from '@polkadot/ui-keyring';
+import { useMemo } from 'react';
+
+export const useAccountAvailable = (accountId: string): boolean | undefined =>
```

Comment:
Yes that's a bit ugly here as the form validation is not as elaborate as it could be. 

The intention to leave `true/false/undefined` was to return a value if `accountId` is not yet set. If the `accountId` is `""` as on the first render, it would return `false`, resulting in the error to be shown. If the `accountId` is not valid in the first place I don't want to show the user an error that the `accountId` is not available. 

`true` => account is valid and available
`false`=> account is valid but unavailable
`undefined` => account is not even valid, so can't say 🤷
(Validity is not checked in the hook just empty string, would be an option to add it but than it would be doubled as only valid options are available by the `AccountSelect` and if new ones are created, they will be checked here: https://github.com/paritytech/contracts-ui/blob/4aa060ed75f432084544c89723a7cf8650b82978/src/ui/components/common/Dropdown.tsx#L104-L106)

That's why I'm checking explicitly for `false` in these lines, instead of using `!isAccountAvailable`, as this would turn `undefined` into `true`.

https://github.com/paritytech/contracts-ui/blob/4aa060ed75f432084544c89723a7cf8650b82978/src/ui/components/instantiate/Step1.tsx#L95

https://github.com/paritytech/contracts-ui/blob/4aa060ed75f432084544c89723a7cf8650b82978/src/ui/components/contract/Interact.tsx#L199

A version either way `useAccountUnavailable` or `useAccountAvailable` returning just `true`/`false` would not be able to distinguish between the invalid/empty `AccountId` resulting in a flash of the error message on first render. 

https://github.com/paritytech/contracts-ui/assets/839848/e6a4985d-e55e-4bb3-ade5-92c5991db89b

All in all I agree it's not he most beautiful way of handling it though. Stuff like this should only be validated if the "Form" is dirty and not enforced on initial values, however tracking if the `AccountSelect` is dirty felt a bit too much.